app/models/site.rb

```diff
@@ -11,11 +11,27 @@ class Site < ActiveRecord::Base
            :announcement_text=, :marketing_text=, :featured_researcher=,
            to: :content_blocks
 
+  def content_by_name(name)
```

Comment:
Removes the need to modify the Site and ContentBlock models when adding a new block somewhere on the site. DRYs up the repeated methods in ContentBlock, though the researcher methods have some unique behavior.
If ContentBlock is going to be touched anyway, maybe these new content_by_name methods ought to go there instead.